calendar/page.js

```diff
@@ -0,0 +1,286 @@
+    const mappedRecord = grist.mapColumnNames(record, mappings);
+    if (mappedRecord && this.calendarHandler) {
+      this.calendarHandler.selectRecord(mappedRecord);
+    }
+}
+
+// when user change the perspective in the GUI, we want to save it as grist option
+// - rest of logic is in reaction to grist option changed
+async function calendarViewChanges(radiobutton) {
+  await grist.setOption('calendarViewPerspective', radiobutton.value);
+}
+
+
+// when user change a perspective of calendar we want this to be persisted in grist options between sessions.
+// This is the place where we can react to this change and update calendar view, or when new session is started
+// (so we are loading previous settings)
+let onGristSettingsChanged = function(options) {
+  let option = options?.calendarViewPerspective??'week';
+    this.calendarHandler.changeView(option);
+    selectRadioButton(option);
+};
+
+// when user move or resize event on the calendar, we want to update the record in the table
+const onCalendarEventBeingUpdated = async (info) => {
+    if (info.changes?.start || info.changes?.end) {
+      const record =  await grist.fetchSelectedRecord(info.event.id);
+      if (record) {
+        // get all the record data from the event, and update only start and end date
+        const gristEvent = buildGristFlatFormatFromEventObject(info.event)
+        if(info.changes.start) gristEvent.startDate = roundEpochDateToSeconds(info.changes.start.valueOf());
+        if(info.changes.end) gristEvent.endDate = roundEpochDateToSeconds(info.changes.end.valueOf());
+        await upsertGristRecord(gristEvent);
+      }
+    }
+};
+
+// saving events to the table or updating existing one - basing on if ID is present or not in the send event
+async function upsertGristRecord(gristEvent){
+    const eventInValidFormat = convertEventToGristTableFormat(gristEvent);
+    const table = await grist.getTable();
+    if (gristEvent.id) {
+        await table.update(eventInValidFormat);
+    } else {
+        await table.create(eventInValidFormat);
+    }
+}
+
+// grist expect date in seconds, but calendar is returning it in miliseconds, so we need to convert it
+function roundEpochDateToSeconds(date) {
+  return date/1000;
+}
+
+// conversion between calendar event object and grist flat format (so the one that is returned in onRecords event
+// and can be mapped by grist.mapColumnNamesBack)
+function buildGristFlatFormatFromEventObject(TUIEvent) {
+  const gristEvent = {
+    startDate: roundEpochDateToSeconds(TUIEvent.start?.valueOf()),
+    endDate: roundEpochDateToSeconds(TUIEvent.end?.valueOf()),
+    isAllDay: TUIEvent.isAllday ? 1 : 0,
+    title: TUIEvent.title??"New Event"
+  }
+  if(TUIEvent.id) gristEvent.id = TUIEvent.id;
+  return gristEvent;
+}
+
+// when user select new date range on the calendar, we want to create new record in the table
+const onNewDateBeingSelectedOnCalendar = async (info) => {
```

Comment:
Any particular reason not to write this as `async function onNew...(info) {` like you've been doing for others?